src/OpenLoco/src/Map/TileManager.cpp

```diff
@@ -480,37 +490,515 @@ namespace OpenLoco::World::TileManager
+                break;
+            }
+            case ElementType::tree:
+            {
+                auto* elTree = el.as<TreeElement>();
+                if (elTree == nullptr)
+                {
+                    return;
+                }
+                auto* treeObj = ObjectManager::get<TreeObject>(elTree->treeObjectId());
+                FormatArguments::common(treeObj->name);
+
+                GameCommands::setErrorText(StringIds::string_id_in_the_way);
+                break;
+            }
+            case ElementType::wall:
+                GameCommands::setErrorText(StringIds::object_in_the_way);
+                break;
+            case ElementType::road:
+            {
+                auto* elRoad = el.as<RoadElement>();
+                if (elRoad == nullptr)
+                {
+                    return;
+                }
+                auto* roadObj = ObjectManager::get<RoadObject>(elRoad->roadObjectId());
+                FormatArguments::common(roadObj->name);
+
+                GameCommands::setErrorText(StringIds::string_id_in_the_way);
+                break;
+            }
+            case ElementType::industry:
+            {
+                auto* elIndustry = el.as<IndustryElement>();
+                if (elIndustry == nullptr)
+                {
+                    return;
+                }
+                auto* industry = elIndustry->industry();
+                FormatArguments::common(industry->name, industry->town);
+
+                GameCommands::setErrorText(StringIds::string_id_in_the_way);
+                break;
+            }
+        }
+    }
+
+    // 0x00462BB3
+    // If return true input el not modified
+    // If return false input el set to 0xFFFFFFFF if error text set or problem el if not set
+    static ClearResult companyAboutToBuildCheck(const World::TileElement& el)
+    {
+        auto* elSurface = el.as<SurfaceElement>();
+
+        // 0x00462C02
+        auto returnFunc = [](const CompanyId owner) -> ClearResult {
+            const auto updatingId = GameCommands::getUpdatingCompanyId();
+            if (updatingId == CompanyId::neutral || !CompanyManager::isPlayerCompany(updatingId))
+            {
+                GameCommands::setErrorText(StringIds::another_company_is_about_to_build_here);
+                return ClearResult::collisionWithErrorMessage;
+            }
+            if (owner == updatingId || CompanyManager::isPlayerCompany(owner))
+            {
+                return ClearResult::collsionNoMessage;
+            }
+            auto* company = CompanyManager::get(owner);
+            if ((company->challengeFlags & CompanyFlags::unk2) != CompanyFlags::none)
+            {
+                GameCommands::setErrorText(StringIds::another_company_is_about_to_build_here);
+                return ClearResult::collisionWithErrorMessage;
+            }
+            // Modification from vanilla
```

Comment:
Interesting. Perhaps good to note _what_ has been modified?

Reply:
if i could remember i would ...